File: Source/WebKit2/Shared/sling/ValueCallback.cpp
/*
 * Copyright (C) 2015 Naver Corp. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *  * Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *  * Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
 * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
 * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

#include "config.h"
#include "ValueCallback.h"

#include <ManagedBoolean.h>
#include <ManagedInteger.h>
#include <ManagedLong.h>

namespace WebKit {
namespace Managed {

void ValueCallback::INIT()
{
}

void ValueCallback::onReceiveError(int32_t error)
{
}

void ValueCallback::onReceiveBoolean(bool value)
{
}

void ValueCallback::onReceiveByte(int8_t value)
{
}

void ValueCallback::onReceiveShort(int16_t value)
{
}

void ValueCallback::onReceiveInt(int32_t value)
{
}

void ValueCallback::onReceiveLong(int64_t value)
{
}

void ValueCallback::onReceiveFloat(float value)
{
}

void ValueCallback::onReceiveDouble(double value)
{
}

void ValueCallback::onReceiveObject(std::shared_ptr<void> value)
{
}

void ValueCallback::onReceiveString(const std::string& value)
{
}

}
}

File: Source/WebKit2/Shared/sling/ValueCallback.java
package labs.naver.webkit.api;

import labs.naver.onig.AccessedByNative;
import labs.naver.onig.CalledByNative;
import labs.naver.onig.NativeClassExport;
import labs.naver.onig.NativeNamespace;

@NativeNamespace("WebKit")
@NativeClassExport("WEBKIT_EXPORT")
public class ValueCallback<T extends Object> {

    // Followed from CallbackBase::Error.
    @AccessedByNative
    public static final int ERROR_NONE = 0;
    @AccessedByNative
    public static final int ERROR_UNKNOWN = 1;
    @AccessedByNative
    public static final int ERROR_PROCESS_EXITED = 2;
    @AccessedByNative
    public static final int ERROR_OWNER_WAS_INVALIDATED = 3;

    public void onReceiveValue(T value) {
        assert false;
    }

    @CalledByNative
    private void onReceiveError(int error) {
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveBoolean(boolean value) {
        onReceiveObject(Boolean.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveByte(byte value) {
        onReceiveObject(Byte.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveShort(short value) {
        onReceiveObject(Short.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveInt(int value) {
        onReceiveObject(Integer.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveLong(long value) {
        onReceiveObject(Long.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveFloat(float value) {
        onReceiveObject(Float.valueOf(value));
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveDouble(double value) {
        onReceiveObject(Double.valueOf(value));
    }

    @SuppressWarnings("unchecked")
    @CalledByNative
    private void onReceiveObject(Object value) {
        onReceiveValue((T) value);
    }

    @SuppressWarnings("unused")
    @CalledByNative
    private void onReceiveString(String value) {
        onReceiveObject(value);
    }
}
